src/hdmf/container.py

```diff
@@ -626,7 +626,10 @@ def _generate_html_repr(self, fields, level=0, access_code="", is_field=False):
         if isinstance(fields, dict):
             for key, value in fields.items():
                 current_access_code = f"{access_code}.{key}" if is_field else f"{access_code}['{key}']"
-                html_repr += self._generate_field_html(key, value, level, current_access_code)
+                if hasattr(value, '_generate_field_html'):
+                    html_repr += value._generate_field_html(key, value, level, current_access_code)
+                else:
+                    html_repr += self._generate_field_html(key, value, level, current_access_code)
```

Comment:
This suggests that `_generate_field_html` is intended as a function that a child-class should be able to overwrite. If this is the case, then we should update the docstring for `_generate_field_html` accordingly. 

We should also add a section to https://nwb-overview.readthedocs.io/en/latest/extensions_tutorial/5_custom_api.html to describe how to customize the html representation of containers.

Reply:
updated the docstring, I can make an issue on nwb-overview to add that section to the extension tutorials